fix: corrects a bug in multi-cluster support

jr0d opened the pull request:
The instructions where using redirect url to determine cluster hostname,
    resulting in all clusters having the kommander cluster hostname.

    Made some changes to support a better plugin instructions template.

jr0d committed:
fix: corrects a bug in multi-cluster support

    The instructions where using redirect url to determine cluster hostname,
    resulting in all clusters having the kommander cluster hostname.

    Made some changes to support a better plugin instructions template.

jr0d committed:
fix: add the kommander CA to the instructions

dlipovetsky commented on asyncauth.go:
```diff
@@ -219,6 +219,12 @@ func (config *Config) renderInstructions(w http.ResponseWriter, req *http.Reques
 
+	// Get the Kommander Cluster CA
+	authCAData := ""
+	if config.Clusters[0].K8s_Ca_Pem != "" {
```
Segmentation fault if `len(config.Clusters) < 1`? I see this in code that's not being updated by this PR, so not a new issue, but I'm curious.

jr0d replied:
You are correct, this is me being lazy and pasting bad code, which I probably wrote. In my defence, this "should never happen", the application would bail before getting here if there were no clusters defined. But, if that behaviour changes... then :boom:.

dlipovetsky replied:
Ok, cool. Just checking :joy:

jr0d committed:
fix: address PR comment

    Explicit runtime panic when attempting to index an empty cluster list.
    There is a circular dependency between Config and Cluster which
    warrants a larger refactor.

dlipovetsky commented on dex-auth.go:
```diff
@@ -87,6 +87,13 @@ func (config *Config) getCluster(name string) *Cluster {
 	return nil
 }
 
+func (config *Config) getFirstClusterOrPanic() *Cluster {
```
:100:

dlipovetsky commented:
Just curious.. would it make sense to check that `len(config.Clusters) > 0` when the config is [initialized](https://github.com/mesosphere/dex-k8s-authenticator/blob/mesosphere/main.go#L365)? (I think `.Clusters` remains static after that, but not sure...)

dlipovetsky commented on asyncauth.go:
```diff
@@ -334,22 +347,22 @@ func (config *Config) downloadKubeConfig(w http.ResponseWriter, req *http.Reques
 	kUser := KConfigUser{
 		Name:    profileName,
 		AuthURL: asyncAuthURL,
-		Command: runPath,
```
I didn't see `runPath` constant used anywhere else. Remove it?

jr0d replied:
looks like it's still used here: https://github.com/mesosphere/dex-k8s-authenticator/blob/480d76031ccf7ea49c075163e062c7fd90c8154d/asyncauth.go#L238

dlipovetsky commented on asyncauth.go:
```diff
@@ -191,27 +199,31 @@ func (config *Config) renderInstructions(w http.ResponseWriter, req *http.Reques
 			if parsed.Hostname() == selectCluster {
-				cluster = c
-				found = true
+				cluster = &c
```
nit: `cluster` refers to the value of `c` in the last iteration. It's been a (subtle) source of bugs for me, and for that reason prefer to reference the original value (using the index), i.e., `cluster = &config.Clusters[i]`. In this case, there's no problem because of the `break`.

jr0d replied:
:+1:

jr0d committed:
fix: address PR comments
